infrastructure/simulation-tool/test/operations/changePubKey.test.ts

```diff
@@ -0,0 +1,219 @@
+    resolvePubKeyChanges,
+    ResolvedPubKeyChange
+} from '../../src/operations/changePubKey';
+import { expect, use } from 'chai';
+import { constants } from 'ethers';
+import { RollupWalletGenerator } from '../../src/utils/wallet.utils';
+
+use(sinonChai);
+
+describe('changePubKey', () => {
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    describe('preparePubKeyChange', () => {
+        let senderWallet: RollupWallet;
+        beforeEach(() => {
+            senderWallet = sinon.createStubInstance(RollupWallet);
+        });
+        it('should return object with "from" set to the sender parameter', () => {
+            const expectedSender = senderWallet;
+            const { from } = preparePubKeyChange(expectedSender);
+
+            expect(from).to.deep.equal(expectedSender);
+        });
+
+        it('should return object with "ethAuthType" set to "ECDSA"', () => {
+            const expectedValue = 'ECDSA';
+            const { ethAuthType } = preparePubKeyChange(senderWallet);
+
+            expect(ethAuthType).to.deep.equal(expectedValue);
+        });
+
+        it('should return object with "feeToken" set to zero address', () => {
+            const expectedFeeToken = constants.AddressZero;
+            const { feeToken } = preparePubKeyChange(senderWallet);
+
+            expect(feeToken).to.deep.equal(expectedFeeToken);
+        });
+    });
+
+    describe('generatePubKeyChanges', () => {
+        it(`should return given number of objects`, async () => {
+            const walletGenerator: RollupWalletGenerator = {
+                next: sinon.stub().resolves({ value: sinon.createStubInstance(RollupWallet) })
+            } as unknown as RollupWalletGenerator;
+            const expectedCount = 12;
+            const { length: actualCount } = await generatePubKeyChanges(expectedCount, walletGenerator);
+
+            expect(actualCount).to.equal(expectedCount);
+        });
+
+        it('should only return objects for accounts without signing key already set', async () => {
+            const txCount = 12;
+            const keysSetIndexes = [9, 3, 7];
+            let genCount = 0;
+            const createWalletStub = (i: number) => {
+                if (genCount > txCount + keysSetIndexes.length) {
+                    throw Error('Halting problem detected.');
+                }
+
+                const stub = sinon.createStubInstance(RollupWallet);
+                stub.isSigningKeySet.resolves(keysSetIndexes.includes(i));
+
+                return stub;
+            };
+
+            const walletGenerator: RollupWalletGenerator = {
+                next: sinon.stub().callsFake(() => Promise.resolve({ value: createWalletStub(genCount++) }))
+            } as unknown as RollupWalletGenerator;
+
+            const results = await generatePubKeyChanges(txCount, walletGenerator);
+            const actualResults = await Promise.all(results.map(async ({ from }) => await from.isSigningKeySet()));
```

Comment:
Why do we need both `await Promise.all...` and `await from.isSigningKeySet`?

Reply:
done